web/app/js/components/TapEventTable.jsx

```diff
@@ -0,0 +1,136 @@
+import _ from 'lodash';
+import BaseTable from './BaseTable.jsx';
+import { publicAddressToString } from './util/Utils.js';
+import React from 'react';
+import { Col, Icon, Row } from 'antd';
+
+let tapColumns = [
+  {
+    title: "ID",
+    dataIndex: "base.id"
+  },
+  {
+    title: "Direction",
+    dataIndex: "base.proxyDirection"
+  },
+  {
+    title: "Source",
+    dataIndex: "base.source",
+    render: d => !d ? null : <span>{publicAddressToString(_.get(d, "ip.ipv4"), d.port)}</span>
+  },
+  {
+    title: "Destination",
+    dataIndex: "base.destination",
+    render: d => !d ? null : <span>{publicAddressToString(_.get(d, "ip.ipv4"), d.port)}</span>
+  },
+  {
+    title: "TLS",
+    dataIndex: "base.tls"
+  },
+  {
+    title: "Request Init",
+    children: [
```

Comment:
It's interesting to see this nesting in the web UI:

![image](https://user-images.githubusercontent.com/9226/43556169-34dbb17e-95b3-11e8-8ff6-553ef0885ccb.png)

But I think ultimately whether we get the data from `RequestInt`, `ResponseInit`, or `ResponseEnd` feels like an implementation detail. I think my preference would be to remove those groupings and flatten all of the column headers.

On a related note, seeing `Response Init Latency` and `Response End Latency` in the same table is a bit confusing. Maybe we could just pick one and display it by default?

Reply:
Hmmm what about for streams and long running requests? It might not be obvious why some fields in the table haven't shown up yet? I find it a bit helpful to figure out what "stage" we're in wrt this, but I'm sure we can find other ways to also express this.

![screen shot 2018-08-01 at 3 36 37 pm](https://user-images.githubusercontent.com/549258/43600785-6a9ee01e-9640-11e8-9b14-c067c238bb8c.png)

For resp init/end latency, I'm not sure which of them we'll be relying on more yet. I'd like to keep them around for now until we can play around with it a bit more, and then play around with the display of that data, if that's ok.

Reply:
That works for me. I like the approach of leaving all of the data and fields in the table for now, and then playing around more with display later.